src/java/org/apache/cassandra/streaming/async/StreamCompressionSerializer.java

```diff
@@ -94,15 +87,15 @@ public ByteBuffer deserialize(LZ4FastDecompressor decompressor, DataInputPlus in
 
         if (in instanceof ReadableByteChannel)
         {
-            ByteBuffer compressed = BufferPool.get(compressedLength);
+            ByteBuffer compressed = ByteBuffer.allocateDirect(compressedLength);
             ByteBuffer uncompressed = null;
             try
             {
                 int readLength = ((ReadableByteChannel) in).read(compressed);
                 assert readLength == compressed.position();
                 compressed.flip();
 
-                uncompressed = BufferPool.get(uncompressedLength);
+                uncompressed = ByteBuffer.allocateDirect(uncompressedLength);
```

Comment:
What about Netty's allocator? That might be faster since it's supposed to be a jemalloc like implementation. allocateDirect will have to create a cleaner and cleaners can be kind of a shit show and are a global synchronization point. Maybe it's fine since only streaming will be doing this. I don't know.